src/server/system_services/bucket_server.js

```diff
@@ -36,6 +36,9 @@ const chunk_config_utils = require('../utils/chunk_config_utils');
 const NetStorage = require('../../util/NetStorageKit-Node-master/lib/netstorage');
 const { OP_NAME_TO_ACTION } = require('../../endpoint/s3/s3_utils');
 const path = require('path');
+const KeysSemaphore = require('../../util/keys_semaphore');
+const bucket_create_semaphore = new KeysSemaphore(1);
+const bucket_delete_semaphore = new KeysSemaphore(1);
```

Comment:
shouldn't there be just one key semaphore? no reason to allow deletion while creation is in progress

Reply:
@jackyalbo And I talked about that,
Why not? Why we should not allow deletion while creation is in progress?

Reply:
why should we? if the bucket creation is in progress, the behavior of the deletion is inconsistent. it might work or not. 
in any case, it's an edge case. but if we already surrounding delete_bucket with a key semaphore I think we should use the same as create

Reply:
ok, after talking a bit with @dannyzaken we decided to change to one semaphore

Reply:
Yes, It is an edge case, this is only when trying to delete a bucket with the same name as one that is currently being created.